frontend/projects/upgrade/src/app/features/dashboard/home/components/metrics/metrics.component.ts

```diff
@@ -457,11 +459,14 @@ export class MonitoredMetricsComponent implements OnInit, OnChanges, OnDestroy {
           (query, index) => {
```

Comment:
if i'm reading this right, this isn't a `map` it's just a `forEach` because we never return anything.

Reply:
correct

Reply:
okay my bad, I think I got confused because sometimes it doesn't return anything when `keys` is falsey (well, it does, it returns `undefined` since the `return` is still implied even when removed), so we would be potentially building an array of `undefined`s. Maybe that just works though, the code has lots of conditionals in it, trying my best to follow, I wonder if in my mind I'm trying to turn it into a `forEach` where `if (keys)` and `keys.length` are both true after filtering, then `push` onto the `monitoredMetricsFormData.queries` array, otherwise do nothing... but if this works in test cases, i'm okay with it.